iOSDeviceManager/Utilities/DeviceUtils.m

```diff
@@ -100,52 +126,47 @@ + (NSString *)findDeviceIDByName:(NSString *)name {
-    NSString *otherSimDeviceName = [otherSim name];
-
-    if ([simVersion isGreaterThan:otherSimVersion]) {
-        return NSOrderedDescending;
-    } else if ([simVersion isEqual:otherSimVersion]) {
-        if ([simDeviceName containsString:@"iPhone"] && [otherSimDeviceName containsString:@"iPhone"]) {
-            NSCharacterSet *nonDigitCharacterSet = [[NSCharacterSet decimalDigitCharacterSet] invertedSet];
-            NSString *simNumber = [[simDeviceName componentsSeparatedByCharactersInSet:nonDigitCharacterSet] componentsJoinedByString:@""];
-            NSString *otherSimNumber = [[otherSimDeviceName componentsSeparatedByCharactersInSet:nonDigitCharacterSet] componentsJoinedByString:@""];
-            if (simNumber.length == 0) {
-                return NSOrderedAscending;
-            }
-            if (otherSimNumber.length == 0) {
-                return NSOrderedSame;
-            }
-            if (fabs(simNumber.doubleValue - otherSimNumber.doubleValue) < EPSILON) {
-                // Handle things like 6S vs 6
-                NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:@".+\\d+[Ss]" options:0 error:nil];
-                NSUInteger simIsS = [regex numberOfMatchesInString:simDeviceName options:0 range:NSMakeRange(0, [simDeviceName length])];
-                NSUInteger otherSimIsS = [regex numberOfMatchesInString:otherSimDeviceName options:0 range:NSMakeRange(0, [otherSimDeviceName length])];
-
-                if (simIsS > 0 && otherSimIsS == 0) {
-                    return NSOrderedDescending;
-                }
-            } else if ((simNumber.doubleValue - otherSimNumber.doubleValue) > EPSILON) {
-                return NSOrderedDescending;
-            } else {
-                return NSOrderedAscending;
-            }
-        } else if ([simDeviceName containsString:@"iPhone"] && ![otherSimDeviceName containsString:@"iPhone"]) {
-            return NSOrderedDescending;
+    // step 1. define desired simulator model and runtime
+    int xcode_major;
+    int xcode_minor;
+    XcodeVersion(&xcode_major, &xcode_minor);
+    // TODO: runtime versionwill be used in iOS version comparision
+    // int major=xcode_major+2;
+    // int minor=xcode_minor;
+    
+    NSString *defaultModel;
+    
+    if (xcode_major == 10) {
+        defaultModel=@"XS";
+    }else{
+        defaultModel=[NSString stringWithFormat:@"%d", xcode_major-1];
+    }
+    
+    // step 2. find FBSimulator with the desired simulators
+    // Re explanation:
+    //   iPhone\\s+  - skip anything before model name
+    //   (\\d+|XS)   - pickup either XS or one of 4..8 model and save into capture group #1
+    NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:@"iPhone\\s+(\\d+|XS)" options:0 error:nil];
+    
+    // while we do not have iOS information will assign each matched simulator
+    // to "def" variable. The last one will be returned
+    FBSimulator *def = nil;
```

Comment:
Use more descriptive variable names.  I would use `simulator` here.

Reply:
renamed as `defaultSimulatorCandidate` in order to emphasize we will not pickup the first matched model but will traverse the list till the most recent iOS